examples/history_vehicles_replacement_for_imitation_learning.py

```diff
@@ -14,50 +14,52 @@
 from smarts.core.sensors import Observation
 from smarts.core.smarts import SMARTS
 from smarts.core.traffic_history_provider import TrafficHistoryProvider
-from smarts.core.utils.math import min_angles_difference_signed
 
 logging.basicConfig(level=logging.INFO)
 
 
-class PlaceholderAgent(Agent):
+class ReplayCheckerAgent(Agent):
     """This is just a place holder such the example code here has a real Agent to work with.
+    This agent checks that the action space is working 'as expected'.
     In actual use, this would be replaced by an agent based on a trained Imitation Learning model."""
 
-    def __init__(self, initial_speed: float = 15.0):
-        self._initial_speed = initial_speed
-        self._initial_speed_set = False
-
-    @staticmethod
-    def _dist(pose1, pose2):
-        return math.sqrt((pose1[0] - pose2[0]) ** 2 + (pose1[1] - pose2[1]) ** 2)
+    def __init__(self, timestep_sec: float):
+        self._timestep_sec = timestep_sec
+
+    def load_data_for_vehicle(
+        self, vehicle_id: str, scenario: Scenario, time_offset: float
+    ):
+        self._vehicle_id = vehicle_id  # for debugging
+        self._time_offset = time_offset
+        datafile = f"data_{scenario.name}_{scenario.traffic_history.name}_Agent-history-vehicle-{vehicle_id}.pkl"
+        # We read actions from a datafile previously-generated by the
+        # observation_collection_for_imitation_learning.py script.
+        # This allows us to test the action space to ensure that it
+        # can recreate the original behaviour.
+        with open(datafile, "rb") as pf:
+            self._data = pickle.load(pf)
```

Comment:
Yeah, and also real imitation learning users will neither be using this "fake" agent, nor using the code in the observations script to write the `.pkl` files in the first place (because they will presumably modify that example to generate the data as they need it for their model training.